d1985r0.md

```diff
@@ -414,6 +423,163 @@ reflection.
 TODO. Concievably simplifies the implementation and makes it far more
 general.
 
+# Covariance and Contravariance
+
+Consider the two concepts:
+
+```cpp
+template <typename T> concept A = true;
+// B subsumes A
+template <typename T> concept B = A<T> && true;
+```
+
+::: cmptable
+
+## Covariance
+```cpp
+auto returns_a() -> is_a;
+auto returns_b() -> is_b;
+auto f() {
+    // OK, requirement less constrained than returned value
+    is_a x = returns_b();
+    // Error, requirement stricter than returned value
+    is_b y = returns_a();
+}
+```
+
+## Contravariance
+```cpp
+template <template <B> typename f>
+using puts_b = void;
+template <template <A> typename f>
+using puts_a = void;
+template <B> using takes_b = void;
+template <A> using takes_a = void;
+using x = puts_b<takes_a>; // ok
+// Error, constraint mismatch (gcc)
+// clang accepts in error
+using w = puts_a<takes_b>;
+```
+
+:::
+
+We are used to the covariant case, but the usage of contravariant cases is
+not as common. The issue with `puts_a<takes_b>` is that `puts_a` requires a
+_more general_ function than one that just accepts `B`s.
```

Comment:
I am not sure if that is just me ;-) but I find the wording "wider interface" is easier to understand than "more general function".

Reply:
awesome, fixed.